Framework/HistogramData/test/FixedLengthVectorTest.h

```diff
@@ -107,6 +107,30 @@ class FixedLengthVectorTest : public CxxTest::TestSuite {
     TS_ASSERT_EQUALS(testee[2], 1.0);
   }
 
+  void test_range_assignment() {
+    std::vector<double> src{3.6, 9.7, 8.5};
+    FixedLengthVectorTester dest;
+
+    dest.assign(src.cbegin(), src.cend());
+
+    TS_ASSERT_EQUALS(dest.size(), 3);
+    TS_ASSERT_EQUALS(dest[0], src[0]);
+    TS_ASSERT_EQUALS(dest[1], src[1]);
+    TS_ASSERT_EQUALS(dest[2], src[2]);
+  }
```

Comment:
I overlooked this earlier: Of course you need to verify the assignment size in the case of `FixedLengthVector` (as the name implies). See other cases like `operator=`.